### PR TITLE
1.20.5 Protocol

### PR DESCRIPTION
**Pull request by forester302**
Update protocol to 1.20.5 as specified in https://wiki.vg/index.php?title=Pre-release_protocol&oldid=19287

### CODE REVIEW

**Comment by extremeheat** [comment]
There are several things missing. For example any places where item stacks are written now have optional slots (like [ServerboundSetCreativeModeSlotPacket](https://github.com/extremeheat/extracted_minecraft_data/blame/client1.20.5/client/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java#L14), [ServerboundContainerClickPacket](https://github.com/extremeheat/extracted_minecraft_data/blame/client1.20.5/client/net/minecraft/network/protocol/game/ServerboundContainerClickPacket.java#L59), etc). Also [ServerboundContainerButtonClickPacket ](https://github.com/extremeheat/extracted_minecraft_data/blame/client1.20.5/client/net/minecraft/network/protocol/game/ServerboundContainerButtonClickPacket.java#L11) now uses varints over i8's. There are likely more.

File: data/pc/1.20.5/protocol.json [hunk 1]
```diff
@@ -0,0 +1,7648 @@
...
+                          "type": "i8"
+                        },
+                        {
+                          "name": "z",
+                          "type": "i8"
+                        },
+                        {
+                          "name": "direction",
+                          "type": "u8"
+                        },
```
Comment (extremeheat) [inline data/pc/1.20.5/protocol.json:3631]: these might be double, double, float now 
https://github.com/extremeheat/extracted_minecraft_data/blame/client1.20.5/client/net/minecraft/world/item/component/MapDecorations.java#L27-L30

File: data/pc/1.20.5/protocol.json [hunk 2]
```diff
@@ -0,0 +1,7648 @@
...
+            {
+              "name": "global",
+              "type": "bool"
+            }
+          ]
+        ],
+        "packet_world_particles": [
+          "container",
+          [
+            {
```
Comment (extremeheat) [inline data/pc/1.20.5/protocol.json:3352]: particleId was moved https://github.com/extremeheat/extracted_minecraft_data/blame/client1.20.5/client/net/minecraft/network/protocol/game/ClientboundLevelParticlesPacket.java#L52

**Comment by extremeheat** [comment]
Also need item fixed

* ClientboundUpdateRecipesPacket
* ClientboundUpdateAdvancementsPacket.java

File: data/pc/1.20.5/protocol.json [hunk 3]
```diff
@@ -147,1016 +147,610 @@
...
+    ],
+    "max_stack_size_component": [
+      "container",
+      [
+        {
+          "name": "maxStackSize",
+          "type": "varint"
+        }
+      ]
+    ],
```
Comment (extremeheat) [inline data/pc/1.20.5/protocol.json:167]: is there a reason these are not inlined?

**Comment by rom1504** [comment]
continuing at #898

**Pull request closed without merging**